pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -3610,6 +3628,21 @@ private void waitForSegmentsBecomeOnline(String tableNameWithType, Set<String> s
             tableNameWithType, segmentsToCheck));
   }
 
+  public Set<String> getSegmentsFromIdealStateMatchingState(String tableNameWithType,
```

Comment:
I prefer the previous version `getOnlineSegmentsFromIdealState(String tableNameWithType, boolean includeConsuming)` for simplicity and efficiency. It will be very rare to ask for segments other than ONLINE/CONSUMING.

Reply:
Ack